static/static_ip.sh

```diff
@@ -177,14 +178,14 @@ ENTERNS2
     done
 fi
 
-# Check if DNS is set manaully and set variables accordingly
-if [ -n "$NSIP1" ]
-then
-    DNS1="$NSIP1"
-fi
-
-if [ -n "$NSIP2" ]
+# Use the updated DNS IP addresses
+DNSs="$NSIP1"
+DNS1="$NSIP1"
```

Comment:
Have a look at this: https://github.com/nextcloud/vm/blob/master/nextcloud_install_production.sh#L203

The default behaviour is setting Quad9s DNS, and that's how the VM are released. In other words, NSIP1 is never empty.

Point 2: Why do you set the same DNS for both DNS1 and DNS2?

Point 3 Why don't you use DNS2 in the netplan config? 

Point 4: Maybe I'm tired but I'm not following here. Please explain.

Reply:
Point 3, Sorry I can see that you do. It's not used as Fallback in resolved though. Any reason for that?

Reply:
I have not changed the nextcloud_install_production.sh file but I believe it needs updating based on my big comments below.

Reply:
For points 2, I hope I don't set the same DNS for DNS1 and DNS2. I put DNS1 and DNS2 into the DNSs variable and use that for the netpan file

Reply:
Updated code to pass through DNS1 and DNS2 to DNSs if no DNS servers set